src/psyclone/psyir/nodes/node.py

```diff
@@ -1092,6 +1095,10 @@ class or a tuple of classes) is found. This can be used to avoid
             return local_list
```

Comment:
Although this is simple, I find it hard to pass-up the opportunity for efficiency savings. If `depth` is passed into the recursive call to `walk` then it will always be available (if the user has supplied a value) and we can stop walking down a particular branch of the tree as soon as a node's depth exceeds the target depth?

Reply:
If this is a goer then please could you also update the method docstring to mention this functionality.

Reply:
[Now https://github.com/stfc/PSyclone/pull/2377/commits/8c90d967dfc4d2112b9ac05eac408b549a7b83b3]